lib/neo4j/http/object_wrapper.rb

```diff
@@ -9,11 +9,10 @@ class ObjectWrapper
         :label,
         :original_attributes
 
-      def initialize(label:, graph_node_primary_key_name: nil, **attributes)
+      def initialize(label:, primary_key_name: nil, **attributes)
         @original_attributes = (attributes || {}).with_indifferent_access
         @attributes = original_attributes.dup.with_indifferent_access
-        @key_name = graph_node_primary_key_name
-        @key_value = @attributes.delete(key_name)
```

Comment:
I see that we are extracting this in subclasses now, now in the base `ObjectWrapper` class? Is this because Node and Relationship handle this differently? (delete vs dig)? I'm guessing because relationships don't always have a key to upsert on? Sometimes it uses the two nodes as its "identifier" correct?

Reply:
yep, that's exactly right.

Reply:
Also, node deletes the key attribute from the hash and I can forsee cases for relationships where we want that as the primary key, but also want to see it in the attributes on the relationship (so dig instead).